packages/teleport/src/DesktopSession/DesktopSession.tsx

```diff
@@ -153,17 +151,24 @@ function Session(props: PropsWithChildren<State>) {
             ...prevState,
             enabled: false,
           }));
-          setIsRecording(false);
           setIsSharingDirectory(false);
           tdpClient.nuke();
         }}
         userHost={`${username}@${hostname}`}
         clipboardSharingEnabled={clipboardSharingActive}
-        isRecording={isRecording}
         canShareDirectory={canShareDirectory}
         isSharingDirectory={isSharingDirectory}
         onShareDirectory={() => {
-          setIsSharingDirectory(true);
+          window
```

Comment:
Should probably add a check for the existence of `window.showDirectoryPicker` so that this doesn't throw on unsupported browsers. 

And as a little nitpick, I don't like defining functions within the render portion of a component as it bloats the markup and makes it harder to follow.

Reply:
> Should probably add a check for the existence of window.showDirectoryPicker so that this doesn't throw on unsupported browsers.

I left that as a "known issue" for now: https://github.com/gravitational/webapps/issues/959, since we will want to add some error handling and UI message, but I am just focusing on getting the core functionality working.

Reply:
Basically I'm trying to charge forward with the core implementation in browsers that actually support it, and then go back at the end and write the code that makes sure the UX isn't totally confusing in browsers that don't.

Reply:
FWIW, I tried it with Firefox and I wasn't able to get to the point where the UI would allow me to trigger this code. The Web UI blocked me from using remote desktops and showed this message:

> Your user role supports clipboard sharing over desktop access, however this feature is only available on chromium based browsers like Brave or Google Chrome. Please switch to a supported browser.

It didn't start the session so there was no risk of this code crashing.